src/Microsoft.Extensions.ML/PredictionEnginePool.cs

```diff
@@ -79,9 +80,11 @@ public ITransformer GetModel()
         /// </param>
         public PredictionEngine<TData, TPrediction> GetPredictionEngine(string modelName)
         {
-            if (_namedPools.ContainsKey(modelName))
+            PoolLoader<TData, TPrediction> existingPool = null;
+            _namedPools.TryGetValue(modelName, out existingPool);
+            if (_namedPools.TryGetValue(modelName, out existingPool))
```

Comment:
💡 All three of these lines can be reduced to just one:

```csharp
if (_namedPools.TryGetValue(modelName, out var existingPool))
```

Reply:
Good to know, thanks Sam

---
In reply to: [357439017](https://github.com/dotnet/machinelearning/pull/4570#discussion_r357439017) [](ancestors = 357439017)